go/parquet/pqarrow/schema.go

```diff
@@ -249,7 +249,7 @@ func structToNode(typ *arrow.StructType, name string, nullable bool, props *parq
 	return schema.NewGroupNode(name, repFromNullable(nullable), children, -1)
 }
 
-func fieldToNode(name string, field arrow.Field, props *parquet.WriterProperties, arrprops ArrowWriterProperties) (schema.Node, error) {
+func fieldToNode(name string, field arrow.Field, props *parquet.WriterProperties, arrprops ArrowWriterProperties, customLogicalType *LogicalType) (schema.Node, error) {
```

Comment:
Maybe instead of adding a function parameter here we could optionally specify the logical type in [ColumnProperties](https://github.com/apache/arrow/blob/ee86db57bb771dbdb5f2fca5890854f27e6aa964/go/parquet/writer_properties.go#L54) which can be accessed per-column through the existing `props` parameter.

Reply:
I think that would be confusing for the `parquet` APIs.  
When writing parquet files, logical types should be specified right when creating the parquet schema.  
This extra parameter is only relevant for `pqarrow` only so I think it should be within the scope of `pqarrow`.  

Perhaps I can pass it down via `pqarrow.ArrowWriterProperties` instead?  
![image](https://github.com/user-attachments/assets/b0e51598-b9ab-4fb3-a368-889ecc9b19e4)

Reply:
^ 88796be

Reply:
^ changed in ef9f2f6